# City of Seattle Staff Demographic Analysis
Mark Fishman, 9/17/19

## Abstract
In this lab, we will be analyzing the demographics for the staff of the City of Seattle. The questions we are trying to answer are: How old/young is the oldest and youngest persons working for the City of Seattle? What is the average age of the workers? What is the lowest wage of an employee? What is the highest wage of an employee? What is the average wage of the employees?

## Dataset Preparation
Our dataset will be extracting
- the ages
- the hourly rates

We got our dataset from https://data.seattle.gov/City-Business/City-of-Seattle-Staff-Demographics/5avq-r9hj

In [33]:
# Opening the file and storing all data in a list
file_data = open('City_of_Seattle_Staff_Demographics.csv', 'r');
all_data = [];
for row in file_data:
    all_data.append(row);
file_data.close;


# Deleting the first row of data
del all_data[0];
# Adding all ages to age_data and all wages to wage_data
age_data = [];
wage_data = [];
for item in range (len(all_data)):
    data_split = all_data[item].split(",");
    age_data.append(int(data_split[3]));
    wage_data.append(float(data_split[4]));    


In this code I extracted all of the .csv file into a list. From there, I extracted everyones age into a list and everyones wage into a list

## Data Modeling

In [35]:
# Setting all variables needed for calculations or other information
min_age = min(age_data);
max_age = max(age_data);
min_wage = min(wage_data);
max_wage = max(wage_data);
count_age = len(age_data);
count_wage = len(wage_data);
sum_age = sum(age_data);
sum_wage = sum(wage_data);

# Finding the average age and wage
average_age = (sum_age / count_age);
average_wage = (sum_wage / count_wage);

# Displaying data found
print("AGE DATA:");
print("The youngest city of Seattle employee is:", min_age, "years old");
print("The oldest city of Seattle employee is:", max_age, "years old");
print("The average city of Seattle employee is: %.2f" % average_age, "years old", '\n');
print("WAGE DATA:");
print("The lowest wage of a city of Seattle employee is: $", min_wage, sep='');
print("The highest wage of a city of Seattle employee is: $%.2f" % max_wage, sep='');
print("The average wage of a city of Seattle employee is: $%.2f" % average_wage, sep='');

AGE DATA:
The youngest city of Seattle employee is: 14 years old
The oldest city of Seattle employee is: 92 years old
The average city of Seattle employee is: 44.53 years old 

WAGE DATA:
The lowest wage of a city of Seattle employee is: $5.53
The highest wage of a city of Seattle employee is: $162.84
The average wage of a city of Seattle employee is: $40.03


To find all of the data for the age and wage, I used the python keywords `min`, `max`, `sum` and `len` to find the minimum, maximum and average age/wage. To find the average, I took the sum of all the ages/wages and divided them by toal amount of people

## Data Analysis and Conclusion

Using this data, I found that the youngest employee is 14 years old, the oldest empoyee is 92 years old and the average employee is 44.53 years old. I also found that the lowest wage of an employee is \\$5.53, the highest wage is \\$163.84 and the average wage is \\$40.03. This lets me come to the conclusion that there is a wide variety of ages of employees who work for the city of Seattle, and that the city of Seattle tends to pay its employees well, with its average wage being roughly \\$40. This dataset has also raised some questions for me, such as:
- Is it legal to only get paid \\$5.53 an hour?
- Is it legal for someone only 14 years old to work?
- What does the 92 year old employee do for work?








## Acknowledgements
- Ms. Sconyers helped me by answering lots of my questions and I also looked at her code to figure out how to do things such as opening a file, or splitting the data by each comma
- Raj helped me by explaining how he completed parts of the assignment that I didn't understand how to complete